Tract template

zhangfanmark opened the pull request:
The updated template copies input data to output.

zhangfanmark commented:
I will update the tutorial when this change is included into the nightly.

ihnorton commented on DMRIModuleTemplates/Tractography_Scripted/TemplateKey.py:
```diff
@@ -128,38 +145,24 @@ def hasImageData(self,volumeNode):
-    fb_numpy = dsa.WrapDataObject(inputFB.GetPolyData())
-
-    # get first tensor key
-    ten_key = next(ifilter(lambda x: "Tensor_" in x, fb_numpy.PointData.keys()), None)
```
@zhangfanmark did you intend to get rid of the part that is doing the example calculation here?

zhangfanmark commented:
Hi Isaiah,

Yes, this part can be added in the tutorial. We asked them to do this by copying and pasting.

Regards,
Fan

On Oct 5, 2018, at 14:39, Isaiah Norton <notifications@github.com<mailto:notifications@github.com>> wrote:


        External Email - Use Caution

@ihnorton commented on this pull request.

________________________________

In DMRIModuleTemplates/Tractography_Scripted/TemplateKey.py<https://github.com/SlicerDMRI/SlicerDMRI/pull/118#discussion_r223102608>:

>        logging.warn("Missing PolyData")

-    # get proxy object to access underlying array
-    fb_numpy = dsa.WrapDataObject(inputFB.GetPolyData())
-
-    # get first tensor key
-    ten_key = next(ifilter(lambda x: "Tensor_" in x, fb_numpy.PointData.keys()), None)


@zhangfanmark<https://github.com/zhangfanmark> did you intend to get rid of the part that is doing the example calculation here?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/SlicerDMRI/SlicerDMRI/pull/118#pullrequestreview-162156596>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AHfdVvwRmpMh587KU9TmnETJ4PXy6oyDks5uh6ZcgaJpZM4XKr7F>.


The information in this e-mail is intended only for the person to whom it is
addressed. If you believe this e-mail was sent to you in error and the e-mail
contains patient information, please contact the Partners Compliance HelpLine at
http://www.partners.org/complianceline . If the e-mail was sent to you in error
but does not contain patient information, please contact the sender and properly
dispose of the e-mail.

ljod commented:
But it's important the template should do some example calculation. Not everyone will start from the tutorial. Any template should be fully functional as an example. Is there some example calculation in the template now? Thanks!

________________________________
From: Fan Zhang <notifications@github.com>
Sent: Friday, October 5, 2018 7:04:56 PM
To: SlicerDMRI/SlicerDMRI
Cc: Subscribed
Subject: Re: [SlicerDMRI/SlicerDMRI] Tract template (#118)


        External Email - Use Caution

Hi Isaiah,

Yes, this part can be added in the tutorial. We asked them to do this by copying and pasting.

Regards,
Fan

On Oct 5, 2018, at 14:39, Isaiah Norton <notifications@github.com<mailto:notifications@github.com>> wrote:


External Email - Use Caution

@ihnorton commented on this pull request.

________________________________

In DMRIModuleTemplates/Tractography_Scripted/TemplateKey.py<https://github.com/SlicerDMRI/SlicerDMRI/pull/118#discussion_r223102608>:

> logging.warn("Missing PolyData")

- # get proxy object to access underlying array
- fb_numpy = dsa.WrapDataObject(inputFB.GetPolyData())
-
- # get first tensor key
- ten_key = next(ifilter(lambda x: "Tensor_" in x, fb_numpy.PointData.keys()), None)


@zhangfanmark<https://github.com/zhangfanmark> did you intend to get rid of the part that is doing the example calculation here?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/SlicerDMRI/SlicerDMRI/pull/118#pullrequestreview-162156596>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AHfdVvwRmpMh587KU9TmnETJ4PXy6oyDks5uh6ZcgaJpZM4XKr7F>.


The information in this e-mail is intended only for the person to whom it is
addressed. If you believe this e-mail was sent to you in error and the e-mail
contains patient information, please contact the Partners Compliance HelpLine at
http://www.partners.org/complianceline . If the e-mail was sent to you in error
but does not contain patient information, please contact the sender and properly
dispose of the e-mail.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/SlicerDMRI/SlicerDMRI/pull/118#issuecomment-427521113>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AEgM95NiGb3vJzAzdbFYik94z2fPcInqks5uh-WYgaJpZM4XKr7F>.

zhangfanmark commented:
Hi Isaiah,

I have updated the coed to include the calculation of MD and display the output fibers colored by the calculated MD. Could you take a look?

Thanks!

Regards,
Fan

On Oct 6, 2018, at 7:02 AM, ljod <notifications@github.com<mailto:notifications@github.com>> wrote:


        External Email - Use Caution

But it's important the template should do some example calculation. Not everyone will start from the tutorial. Any template should be fully functional as an example. Is there some example calculation in the template now? Thanks!

________________________________
From: Fan Zhang <notifications@github.com<mailto:notifications@github.com>>
Sent: Friday, October 5, 2018 7:04:56 PM
To: SlicerDMRI/SlicerDMRI
Cc: Subscribed
Subject: Re: [SlicerDMRI/SlicerDMRI] Tract template (#118)


External Email - Use Caution

Hi Isaiah,

Yes, this part can be added in the tutorial. We asked them to do this by copying and pasting.

Regards,
Fan

On Oct 5, 2018, at 14:39, Isaiah Norton <notifications@github.com<mailto:notifications@github.com><mailto:notifications@github.com>> wrote:


External Email - Use Caution

@ihnorton commented on this pull request.

________________________________

In DMRIModuleTemplates/Tractography_Scripted/TemplateKey.py<https://github.com/SlicerDMRI/SlicerDMRI/pull/118#discussion_r223102608>:

> logging.warn("Missing PolyData")

- # get proxy object to access underlying array
- fb_numpy = dsa.WrapDataObject(inputFB.GetPolyData())
-
- # get first tensor key
- ten_key = next(ifilter(lambda x: "Tensor_" in x, fb_numpy.PointData.keys()), None)


@zhangfanmark<https://github.com/zhangfanmark> did you intend to get rid of the part that is doing the example calculation here?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/SlicerDMRI/SlicerDMRI/pull/118#pullrequestreview-162156596>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AHfdVvwRmpMh587KU9TmnETJ4PXy6oyDks5uh6ZcgaJpZM4XKr7F>.


The information in this e-mail is intended only for the person to whom it is
addressed. If you believe this e-mail was sent to you in error and the e-mail
contains patient information, please contact the Partners Compliance HelpLine at
http://www.partners.org/complianceline . If the e-mail was sent to you in error
but does not contain patient information, please contact the sender and properly
dispose of the e-mail.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/SlicerDMRI/SlicerDMRI/pull/118#issuecomment-427521113>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AEgM95NiGb3vJzAzdbFYik94z2fPcInqks5uh-WYgaJpZM4XKr7F>.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/SlicerDMRI/SlicerDMRI/pull/118#issuecomment-427565121>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AHfdVusRoDCLew0drLnsu_O84MvtE9W9ks5uiI3RgaJpZM4XKr7F>.

ihnorton commented:
Thanks!
